docs/painless/painless-guide/painless-execute-script.asciidoc

```diff
@@ -154,13 +249,560 @@ POST /_scripts/painless/_execute
+is `28800000` milliseconds, which equates to eight hours. We'll assume that this
+is an approximate time required to write a complete page for a book.
+
+The script then assesses the `author` and either divides or multiplies the
+`pageTime` value. If the author is `Robert A. Heinlein`, the script divides by
+two. Because `Alastair Reynolds` writes much slower, the script multiplies
+`pageTime` by two if that's the matching author.
+
+Then, the script subtracts the release date value (in milliseconds) from the
+calculation of `pageTime` times the `page_count` to determine approximately when
+the author began writing the book.
+
+[source,console]
+----
+POST /_scripts/painless/_execute
+{
+  "script": {
+    "source": """
+    String author = doc['author'].value;
+    long pageTime = 28800000;
+    if (author == 'Robert A. Heinlein') {
+      pageTime /= 2;
+    } else if (author == 'Alastair Reynolds') {
+      pageTime *= 2;
+    }
+    emit(doc['release_date'].value.toInstant().toEpochMilli() - pageTime * doc['page_count'].value);
+    """
+  },
+  "context": "date_field",
+  "context_setup": {
+    "index": "my-index-000001",
+    "document": {
+      "name": "Revelation Space",
+      "author": "Alastair Reynolds",
+      "release_date": "2000-03-15",
+      "page_count": 585
+    }
+  }
+}
+----
+//TEST[continued]
+
+In this case, the author is Alastair Reynolds. Based on a release date of
+`2000-03-15`, the script calculates that the author started writing
+`Revelation Space` on 19 February 1999. Writing a 585 page book in just over one
+year is pretty impressive!
+
+[source,console-result]
+----
+{
+  "result" : [
+    "1999-02-19T00:00:00.000Z"
+  ]
+}
+----
+
+[[painless-runtime-double]]
+===== `double_field`
+Use the `double_field` runtime context for {ref}/number.html[numeric data] of type
+`double`. For example, let's say you have sensor data that includes a `voltage`
+field with values like 5.6. You need to multiply this value, but only for
+sensors that match a specific model number.
+
+Add the following fields to your index mapping. The `voltage` field is a 
+sub-field of the `measures` object.
+
+[source,console]
+----
+PUT my-index-000001
+{
+  "mappings": {
+    "properties": {
+      "@timestamp": {
+        "type": "date"
+      },
+      "model_number": {
+        "type": "keyword"
+      },
+      "measures": {
+        "properties": {
+          "voltage": {
+            "type": "double"
+          }
+        }
+      }
+    }
+  }
+}
+----
+
+The following script matches on any documents where the `model_number` equals
+`QVKC92Q`, and then multiplies the `voltage` value by `1.7`. This script is
+useful when you want to select specific documents and only operate on values
+that match the specified criteria.
+
+[source,console]
+----
+POST /_scripts/painless/_execute
+{
+  "script": {
+    "source": """if (doc['model_number'].value.equals('QVKC92Q'))
+        {emit(1.7 * params._source['measures']['voltage']);}
```

Comment:
We should probably be consistent in how we format the scripts on this page - I'd prefer taller rather than wider even though its hard to get everything on a page. I find it nicer to work with scripts that are formatted that way.

Reply:
Good callout! I'll standardize the format throughout.